[datadog_incident_type] Add configuration attribute (#3942)
DataDog/terraform-provider-datadog: docs/resources/incident_type.md

### Optional

> - `configuration` (Attributes) The incident type's behavior settings. Any field left unset is managed by the server and returned as its server-side value. This block is applied after creation via a separate update call, since the create endpoint does not accept configuration. (see [below for nested schema](#nestedatt--configuration))

Reviewer: ```suggestion
- `configuration` (Attributes) The incident type's behavior settings. Any field left unset takes its server-side default. This block is applied in a separate call after the incident type is created. (see [below for nested schema](#nestedatt--configuration))
```

Reviewer: Suggesting to clarify and trim this down to just what a user needs to know.

Source: `resource_datadog_incident_type.go:105`

Author: Applied this wording to the configuration block description in f8a9a3c; docs regenerated.

- `description` (String) Description of the incident type. The description can have a maximum of 512 characters.
- `is_default` (Boolean) Whether this incident type is the default type.

### Read-Only

- `id` (String) The ID of the incident type.

<a id="nestedatt--configuration"></a>
### Nested Schema for `configuration`

Optional:

- `allow_incident_deletion` (Boolean) Whether incidents of this type can be deleted. Defaults to `false`.
- `allow_workflows` (Boolean) Whether automation workflows can be triggered for incidents of this type. Defaults to `true`.

> - `create_message` (String) An optional message shown to users when they declare an incident of this type.

Reviewer: ```suggestion
- `create_message` (String) An optional message shown to users when they declare an incident of this type. Defaults to an empty string.
```

Reviewer: Suggesting to add the default here for consistency.

Source: `resource_datadog_incident_type.go:149`

Author: Applied — added "Defaults to an empty string." to create_message in f8a9a3c; docs regenerated.
